pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -19,12 +19,175 @@ package kuberuntime
+	linuxConfig := &runtimeApi.LinuxContainerConfig{
+		Resources: &runtimeApi.LinuxContainerResources{},
+	}
+
+	// set linux container resources
+	var cpuShares int64
+	cpuRequest := container.Resources.Requests.Cpu()
+	cpuLimit := container.Resources.Limits.Cpu()
+	memoryLimit := container.Resources.Limits.Memory().Value()
+	if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+		cpuShares = milliCPUToShares(cpuLimit.MilliValue())
+	} else {
+		// if cpuRequest.Amount is nil, then milliCPUToShares will return the minimal number
+		// of CPU shares.
+		cpuShares = milliCPUToShares(cpuRequest.MilliValue())
+	}
+	linuxConfig.Resources.CpuShares = &cpuShares
+	if memoryLimit != 0 {
+		linuxConfig.Resources.MemoryLimitInBytes = &memoryLimit
+	}
+	if m.cpuCFSQuota {
+		// if cpuLimit.Amount is nil, then the appropriate default value is returned
+		// to allow full usage of cpu resource.
+		cpuQuota, cpuPeriod := milliCPUToQuota(cpuLimit.MilliValue())
+		linuxConfig.Resources.CpuQuota = &cpuQuota
+		linuxConfig.Resources.CpuPeriod = &cpuPeriod
+	}
+
+	// set security context options
+	if container.SecurityContext != nil {
+		securityContext := container.SecurityContext
+		if securityContext.Capabilities != nil {
+			linuxConfig.Capabilities = &runtimeApi.Capability{
+				AddCapabilities:  make([]string, 0, len(securityContext.Capabilities.Add)),
+				DropCapabilities: make([]string, 0, len(securityContext.Capabilities.Drop)),
+			}
+			for index, value := range securityContext.Capabilities.Add {
+				linuxConfig.Capabilities.AddCapabilities[index] = string(value)
+			}
+			for index, value := range securityContext.Capabilities.Drop {
+				linuxConfig.Capabilities.DropCapabilities[index] = string(value)
+			}
+		}
+
+		if securityContext.SELinuxOptions != nil {
+			linuxConfig.SelinuxOptions = &runtimeApi.SELinuxOption{
+				User:  &securityContext.SELinuxOptions.User,
+				Role:  &securityContext.SELinuxOptions.Role,
+				Type:  &securityContext.SELinuxOptions.Type,
+				Level: &securityContext.SELinuxOptions.Level,
+			}
+		}
+	}
+
+	return linuxConfig
+}
+
+// makeMounts generates container volume mounts for kubelet runtime api.
+func makeMounts(cid string, opts *kubecontainer.RunContainerOptions, container *api.Container, podHasSELinuxLabel bool) []*runtimeApi.Mount {
+	volumeMounts := []*runtimeApi.Mount{}
+
+	for idx := range opts.Mounts {
+		v := opts.Mounts[idx]
+		m := &runtimeApi.Mount{
+			Name:          &v.Name,
+			HostPath:      &v.HostPath,
+			ContainerPath: &v.ContainerPath,
+			Readonly:      &v.ReadOnly,
+		}
+		if podHasSELinuxLabel && v.SELinuxRelabel {
+			m.SelinuxRelabel = &v.SELinuxRelabel
+		}
+
+		volumeMounts = append(volumeMounts, m)
+	}
+
+	// The reason we create and mount the log file in here (not in kubelet) is because
+	// the file's location depends on the ID of the container, and we need to create and
```

Comment:
Now that `create` and `start` are split into two different states, I think kubelet actually can handle creating this mount.